tests/api/api-resources.test.js

```diff
@@ -0,0 +1,131 @@
+// Copyright Contributors to the Open Cluster Management project
+
+jest.retryTimes(global.retry)
+
+const { fail } = require('assert');
+const { performance } = require('perf_hooks');
+
+const squad = require('../../config').get('squadName')
+const {
+  getKubeConfig,
+  getSearchApiRoute,
+  getToken,
+  searchQueryBuilder,
+  sendRequest,
+} = require('../common-lib/clusterAccess')
+
+const {
+  fetchAPIResourcesWithListWatchMethods,
+  formatResourcesFromSearch,
+  formatFilters,
+  getResourcesFromOC,
+  getClusterList,
+  shouldUseAPIGroup
+} = require('../common-lib/index')
+
+const { sleep } = require('../common-lib/sleep')
+
+// Set list to ignore resources that aren't being collected by Search.
+const ignoreKindResourceList = ['clusterclaim', 'event']
+
+// Set list of resources that require filtering by api group.
+const requireAPIGroup = []
+
+/**
+ * Base test for kubernetes
+ * @param {*} kind
+ * @param {*} apigroup
+ * @param {*} cluster
+ * @param {*} user
+ * @param {*} namespace
+ */
+function baseTest(kind, apigroup, cluster = { type: 'hub', name: 'local-cluster' }, user = 'kubeadmin', namespace = '--all-namespaces') {
+  var runTest = test
+
+  try {
+    var expectedResources = getResourcesFromOC(kind, apigroup, namespace, cluster)
+  } catch (err) {
+    runTest = test.skip
+  }
+
+  runTest(
+    `[P2][Sev2][${squad}] verify data integrity for resource property: ${!apigroup.useAPIGroup ? kind : `${kind}.${apigroup.name}`}`,
+    async () => {
+      const filters = formatFilters(kind, apigroup, namespace, cluster)
+
+      // Fetch data from the search api.
+      var query = searchQueryBuilder({ filters })
+
+      var startTime = performance.now()
+      var resp = await sendRequest(query, token)
+      var endTime = performance.now()
+      var totalElapsedTime = endTime - startTime
+
+      var searchResources = formatResourcesFromSearch(resp)
+      
+      if (totalElapsedTime > 30000) {
+        fail(`Search required more than 30 second to return resources for ${kind}. (TotalElapsedTime: ${totalElapsedTime})`)
+      } else if (totalElapsedTime > 1000) {
+        console.warn(`Search required more than 1 second to return resources for ${kind}. (TotalElapsedTime: ${totalElapsedTime})`)
+      }
+
+      if (searchResources.length != expectedResources.length) {
+        console.warn(`Detected incorrect amount of data matches for (${kind}) resources. Retrying test within 5 seconds.`)
+        await sleep(5000)
+
+        // Refresh the list of resources. There's a chance that more resources were created after the previous fetch.
+        expectedResources = getResourcesFromOC(kind, apigroup, namespace, cluster)
+        resp = await sendRequest(query, token)
```

Comment:
Make the function `getResourcesFromOC()` async, then you can resolve these 2 at the same time.
```suggestion
        [expectedResources, resp] = Promise.All(
                getResourcesFromOC(kind, apigroup, namespace, cluster)
                sendRequest(query, token))
```